src/cmds.c

```diff
@@ -810,25 +810,25 @@ static void cmd_pls_bot(struct userrec *u, int idx, char *par)
     return;
   }
 
-  if (strcmp(addr, "")) {
+  if (addr[0]) {
 #ifndef IPV6
  /* Reject IPv6 addresses */
-    if (!inet_pton(AF_INET, addr, saddr)) {
-      for (i = 0; addr[i]; i++) {
-        if (addr[i] == ':') {
-          dprintf(idx, "Invalid IP address format (this Eggdrop "
-            "was compiled without IPv6 support).\n");
-          return;
-        }
+    for (i=0; addr[i]; i++) {
+      if (addr[i] == ':') {
+        dprintf(idx, "Invalid IP address format (this Eggdrop "
+          "was compiled without IPv6 support).\n");
+        return;
       }
     }
 #endif
- /* Check if user forgot address field */
-    if (*addr == '+') {
-      dprintf(idx, "Bot address may not start with a +.\n");
-      return;
-    }
-    for (i=0; i < addr[i]; i++) {
+ /* Check if user forgot address field by checking if argument is completely
+  * numerical, implying a port was provided as the next argument instead.
+  */
+    for (i=0; addr[i]; i++) {
+      if (addr[i] == '+') {
+        dprintf(idx, "Bot address may not contain a +.\n");
+        return;
+      }
```

Comment:
If I may suggest as replacement for the above if-block (and it's copy at line 1155):
```
#define PORT_ARG_CHARS "+/"
if (strchr(PORT_ARG_CHARS, addr[i])) {
  dprintf(idx, "Bot address may not contain a %c.\n", addr[i]);
  break;
}
```
And then remove the ` && (addr[i] != '/')` below (and at line 1159).
The `#define` should go elsewhere of course (and named perhaps different too).

Reasoning: more correct errorreport, sooner breaking of the forloop for '/' and using  a break instead of return to have the usage reporting too.

Reply:
That is some sexy code right there.